src/frontend/src/lib/components/pages/SettingsSignedIn.svelte

```diff
@@ -97,21 +109,20 @@
 	</KeyValuePairInfo>
 </div>
 
-{#if POUH_ENABLED}
-	<div class="mt-8">
+{#if POUH_ENABLED && nonNullish($userProfileStore)}
+	<div class="mt-8" in:fade>
 		<h2 class="text-base mb-6 pb-1">{$i18n.settings.text.credentials_title}</h2>
 
 		<div class="mt-4">
 			<KeyValuePairInfo>
 				<span slot="key" class="font-bold">{$i18n.settings.text.pouh_credential}:</span>
 				<svelte:fragment slot="value">
-					<!-- TODO: Render spinner if no user profile -->
 					{#if $userHasPouhCredential}
-						<output in:fade class="mr-1.5">
+						<output class="mr-1.5">
```

Comment:
the fade is still useful here I think, it's should probably not be removed as it is meant to smoothness the animation when credentials are requested for the first time